test/plots/areaPlotTests.ts

```diff
@@ -176,28 +191,16 @@ describe("Plots", () => {
 
         svg.remove();
       });
-
-      it("skips invalid Datasets", () => {
-        let twoPointDataset = new Plottable.Dataset([{ foo: 0, bar: 1 }, { foo: 1, bar: 2 }]);
-        areaPlot.addDataset(twoPointDataset);
-        let dummyDataset = new Plottable.Dataset([]);
-        let allAreas = areaPlot.selections([twoPointDataset, dummyDataset]);
-        assert.strictEqual(allAreas.size(), 2, "areas/lines retrieved");
-        let selectionData = allAreas.data()[0];
-        assert.deepEqual(selectionData, twoPointDataset.data(), "new dataset data in selection data");
-
-        svg.remove();
-      });
     });
 
-    it("retains original classes when class is projected", () => {
-      let newClassProjector = () => "pink";
-      areaPlot.attr("class", newClassProjector);
-      areaPlot.renderTo(svg);
-      let areaPath = renderArea.select(".area");
-      assert.isTrue(areaPath.classed("pink"));
-      assert.isTrue(areaPath.classed("area"));
-      svg.remove();
-    });
+    function getPointsInArea(plot: Plottable.Plots.Area<any>, points: Plottable.Point[]) {
+      let y0Binding = plot.y0();
+      let firstPoint = points[0];
+      let lastPoint = points[points.length - 1];
+      let ds = plot.datasets()[0];
+      let lasty0 = y0Binding.accessor(lastPoint, points.length - 1, ds);
+      let firsty0 = y0Binding.accessor(firstPoint, 0, ds);
+      return points.concat([{ x: lastPoint.x, y: lasty0 }, { x: firstPoint.x, y: firsty0 }]);
+    };
```

Comment:
> y0 could be different at each data point

Isn't this more reason why this function is incomplete?  If we set a variable `y0` function, does this function still get all the points?